app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/fragments/Transaction.java

```diff
@@ -1,29 +1,254 @@
 package com.aspirebudgetingmobile.aspirebudgeting.fragments;
 
 
+import android.annotation.SuppressLint;
+import android.app.DatePickerDialog;
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.os.AsyncTask;
 import android.os.Bundle;
-
-import androidx.fragment.app.Fragment;
-
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.appcompat.widget.AppCompatSpinner;
+import androidx.fragment.app.Fragment;
 
 import com.aspirebudgetingmobile.aspirebudgeting.R;
+import com.aspirebudgetingmobile.aspirebudgeting.activities.Home;
+import com.aspirebudgetingmobile.aspirebudgeting.interfaces.TransactionCallBack;
+import com.aspirebudgetingmobile.aspirebudgeting.utils.BasicUtils;
+import com.aspirebudgetingmobile.aspirebudgeting.utils.ObjectFactory;
+import com.aspirebudgetingmobile.aspirebudgeting.utils.SheetsManager;
+import com.google.android.material.card.MaterialCardView;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
 
 public class Transaction extends Fragment {
```

Comment:
This class represents a view, please rename it to AddTransactionView or AddTransactionFragment. The name `Transaction` can be misunderstood as a model object.

Reply:
done